Fix #12743 Do not render empty command containers (Menubar, Toolbar, MenuButton,...) [Draft]

gako opened the pull request:
Initial Draft PR regarding not (always) rendering containers that contain no rendered children. 
Fix #12743

tandraschko commented on primefaces/src/main/java/org/primefaces/component/splitbutton/SplitButtonRenderer.java:
```diff
@@ -87,6 +88,8 @@ public void encodeEnd(FacesContext context, UIComponent component) throws IOExce
             model.generateUniqueIds();
         }
 
+        boolean hasOverlay = shouldBeRendered(context, button);
```
can be removed

tandraschko commented on primefaces/src/main/java/org/primefaces/component/toolbar/ToolbarRenderer.java:
```diff
@@ -115,4 +119,19 @@ public void encodeChildren(FacesContext facesContext, UIComponent component) thr
+            for (UIComponent child : toolbar.getChildren()) {
+                if (child.isRendered() && child instanceof ToolbarGroup) {
+                    ToolbarGroup toolbarGroup = (ToolbarGroup) child;
+                    return toolbarGroup.getChildren().stream().filter(me -> !(me instanceof Separator)).anyMatch(UIComponent::isRendered);
```
shouldnt it use
`shouldBeRendered(FacesContext context, MenuElement menuElement)`
because push/pop?

gako replied:
Theoretically yes, but since Toolbar is the only component that is no real Menu, it does not extend BaseMenuRenderer and does not use MenuItems to define it's children, I added a similiar method that's based around UIComponent to ToolbarRenderer itself.

gako committed:
Rebase changes into single commit to clean up changed files.

gako committed:
add migration notes

Commit:
add placeholder for ajax updates.

If special styling is needed xyz-placeholder class can be used for that by external stylesheet.
